### PR TITLE
Fix _load_textdomain_just_in_time warning

### LINKED ISSUE #4085: debug error with calling `_load_textdomain_just_in_time` too early
This has recently started happening, when the entrypoint `/wp-admin/admin-ajax.php` is being called, eg:.

`192.168.56.1 "www.circularrevolution.local" - [18/Mar/2025:13:26:04 +0000] "POST /wp-admin/admin-ajax.php HTTP/1.1" 200 58 "https://www.circularrevolution.local/wp-admin/admin.php?page=wc-settings&tab=checkout&section=stripe&panel=settings" "Mozilla/5.0 (Windows NT 10.0; Win64; x64; rv:136.0) Gecko/20100101 Firefox/136.0" 0.140`

The call stack is:

```
[18-Mar-2025 13:26:04 UTC] PHP Stack trace:
[18-Mar-2025 13:26:04 UTC] PHP   1. {main}() /var/www/website/wordpress/wp-admin/admin-ajax.php:0
[18-Mar-2025 13:26:04 UTC] PHP   2. require_once() /var/www/website/wordpress/wp-admin/admin-ajax.php:22
[18-Mar-2025 13:26:04 UTC] PHP   3. require_once() /var/www/website/wordpress/wp-load.php:55
[18-Mar-2025 13:26:04 UTC] PHP   4. require_once() /var/www/website/wp-config.php:237
[18-Mar-2025 13:26:04 UTC] PHP   5. do_action($hook_name = 'plugins_loaded') /var/www/website/wordpress/wp-settings.php:559
[18-Mar-2025 13:26:04 UTC] PHP   6. WP_Hook->do_action($args = [0 => '']) /var/www/website/wordpress/wp-includes/plugin.php:517
[18-Mar-2025 13:26:04 UTC] PHP   7. WP_Hook->apply_filters($value = '', $args = [0 => '']) /var/www/website/wordpress/wp-includes/class-wp-hook.php:348
[18-Mar-2025 13:26:04 UTC] PHP   8. woocommerce_gateway_stripe_init('') /var/www/website/wordpress/wp-includes/class-wp-hook.php:324
[18-Mar-2025 13:26:04 UTC] PHP   9. woocommerce_gateway_stripe() /var/www/website/wp-content/plugins/woocommerce-gateway-stripe/woocommerce-gateway-stripe.php:843
[18-Mar-2025 13:26:04 UTC] PHP  10. WC_Stripe::get_instance() /var/www/website/wp-content/plugins/woocommerce-gateway-stripe/woocommerce-gateway-stripe.php:823
[18-Mar-2025 13:26:04 UTC] PHP  11. WC_Stripe->__construct() /var/www/website/wp-content/plugins/woocommerce-gateway-stripe/woocommerce-gateway-stripe.php:108
[18-Mar-2025 13:26:04 UTC] PHP  12. WC_Stripe->init() /var/www/website/wp-content/plugins/woocommerce-gateway-stripe/woocommerce-gateway-stripe.php:178
[18-Mar-2025 13:26:04 UTC] PHP  13. WC_Stripe->get_main_stripe_gateway() /var/www/website/wp-content/plugins/woocommerce-gateway-stripe/woocommerce-gateway-stripe.php:315
[18-Mar-2025 13:26:04 UTC] PHP  14. WC_Stripe_UPE_Payment_Gateway->__construct() /var/www/website/wp-content/plugins/woocommerce-gateway-stripe/woocommerce-gateway-stripe.php:772
[18-Mar-2025 13:26:04 UTC] PHP  15. __($text = 'Stripe', $domain = 'woocommerce-gateway-stripe') /var/www/website/wp-content/plugins/woocommerce-gateway-stripe/includes/payment-methods/class-wc-stripe-upe-payment-gateway.php:152
[18-Mar-2025 13:26:04 UTC] PHP  16. translate($text = 'Stripe', $domain = 'woocommerce-gateway-stripe') /var/www/website/wordpress/wp-includes/l10n.php:306
[18-Mar-2025 13:26:04 UTC] PHP  17. get_translations_for_domain($domain = 'woocommerce-gateway-stripe') /var/www/website/wordpress/wp-includes/l10n.php:194
[18-Mar-2025 13:26:04 UTC] PHP  18. _load_textdomain_just_in_time($domain = 'woocommerce-gateway-stripe') /var/www/website/wordpress/wp-includes/l10n.php:1411
[18-Mar-2025 13:26:04 UTC] PHP  19. _doing_it_wrong($function_name = '_load_textdomain_just_in_time', $message = 'Translation loading for the <code>woocommerce-gateway-stripe</code> domain was triggered too early. This is usually an indicator for some code in the plugin or theme running too early. Translations should be loaded at the <code>init</code> action or later.', $version = '6.7.0') /var/www/website/wordpress/wp-includes/l10n.php:1381
[18-Mar-2025 13:26:04 UTC] PHP  20. wp_trigger_error($function_name = '', $message = 'Function _load_textdomain_just_in_time was called <strong>incorrectly</strong>. Translation loading for the <code>woocommerce-gateway-stripe</code> domain was triggered too early. This is usually an indicator for some code in the plugin or theme running too early. Translations should be loaded at the <code>init</code> action or later. Please see <a href="https://developer.wordpress.org/advanced-administration/debug/debug-wordpress/">Debugging in WordPress</a> for more information. (This message was added in', $error_level = *uninitialized*) /var/www/website/wordpress/wp-includes/functions.php:6054
[18-Mar-2025 13:26:04 UTC] PHP  21. trigger_error($message = 'Function _load_textdomain_just_in_time was called <strong>incorrectly</strong>. Translation loading for the <code>woocommerce-gateway-stripe</code> domain was triggered too early. This is usually an indicator for some code in the plugin or theme running too early. Translations should be loaded at the <code>init</code> action or later. Please see <a href="https://developer.wordpress.org/advanced-administration/debug/debug-wordpress/">Debugging in WordPress</a> for more information. (This message was added in', $error_level = 1024) /var/www/website/wordpress/wp-includes/functions.php:6114
```

### PR DESCRIPTION
**Pull request by malithsen**
Fixes #4085, #4140

## Changes proposed in this Pull Request:

This PR fixes a warning that occurs when translations are loaded too early in the WordPress lifecycle. The warning appears as:

```
Function _load_textdomain_just_in_time was called incorrectly. Translation loading for the woocommerce-gateway-stripe domain was triggered too early.
```

The issue was introduced when the Status page updates were added in https://github.com/woocommerce/woocommerce-gateway-stripe/pull/3878, which instantiates the payment gateway during `plugins_loaded`. This caused translations to be loaded before WordPress's text domain infrastructure was fully initialized.
Initially, I considered moving the entire plugin initialization from `plugins_loaded` to `init`. While that fixed the error and should avoid any future instances that could lead to similar errors, it would have been too broad of a change with potential side effects.
Instead, this PR proposes a more targeted fix of only initializing the status page updates during `init`.

This ensures:

- Translations are loaded at the correct time (priority 10)
- Status page is initialized after translations (priority 15)
- Only the Status page functionality is affected,

## Testing instructions

**To reproduce the error:**

1. Switch to develop.
2. Enable debugging by adding `define( 'WP_DEBUG', true );` to your `wp-config.php`
3. Change site language to `English (UK)` via Settings -> General -> Site Language.
4. Go to Dashboards -> Updates. Click on "Re-install version 6.7.2–en_GB"
5. Notice the warning on top of the page.

**Test the fix:**
1. Switch to this branch.
2. Above notice should no longer be visible.

**Test for regressions:**

Follow the instructions in https://github.com/woocommerce/woocommerce-gateway-stripe/pull/3878

---

-   [ ] Covered with tests (or have a good reason not to test in description ☝️)
-   [ ] Tested on mobile (or does not apply)

### Changelog entry

-   [ ] This Pull Request does not require a changelog entry. (Comment required below)

<details>

<summary>Changelog Entry Comment</summary>

#### Comment 

</details>

**Post merge**

-   [ ] Added testing instructions to the [Release Testing Instructions wiki page](https://github.com/woocommerce/woocommerce-gateway-stripe/wiki/Release-Testing-Instructions) (or does not apply)
-   [ ] Added the [needs docs label](https://github.com/woocommerce/woocommerce-gateway-stripe/labels?q=docs) (or does not apply)
-   [ ] Included this PR in the Release Thread scope (or does not apply)

### CODE REVIEW

**Review by Mayisha — APPROVED** [review]
I followed the steps to reproduce but could not reproduce the error in `develop` 🤷 
But the changes look reasonable and it should fix the issue. Approving, :shipit:
